src/main/java/betterquesting/api2/storage/RandomIndexDatabase.java

```diff
@@ -0,0 +1,22 @@
+package betterquesting.api2.storage;
+
+import java.util.Random;
+
+public class RandomIndexDatabase<T> extends AbstractDatabase<T> {
+
+    private final Random random = new Random();
+
+    @Override
+    public synchronized int nextID() {
```

Comment:
this is only called when creating a new quest, but not on other situations in which a quest may be created, such as `ToolboxToolCopy.java#getNextIDs` (although that appears to be a unique case due to its bulk creation).